src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java

```diff
@@ -429,7 +432,7 @@ public String toString() {
 
         @Override
         public boolean delete() {
-            return file.delete();
+            return file.exists() && Files.delete(file);
         }
```

Comment:
I think that is a bad idea, because deleting recursively can happen a lot more efficiently in a JDBC database (with a lot less queries than would happen if you call the Resources.delete method)

Note that there is also a store.remove() function which _is_ recursive (and returns true if the resource didn't exist yet - which is also a contradiction).